### PR TITLE
refactor: add loggers to SpoonClassFileTransformer.java (fix SonarQube)

### PR DESCRIPTION
**Pull request by zielint0**
SonarQube: *Use a logger to log this exception.*

@pvojtechovsky 
Please tell me if I used logging correctly (according to Spoon rules and according to Java best practices).

### CODE REVIEW

**Comment by nharrand** [comment]
Hello @zielint0 

Maybe `launcher.getEnvironment().debugMessage` calls could actually be transformed into Logger calls too actually...

**Comment by monperrus** [comment]
We don't use one logger per class in Spoon. 

But you're right `e.printStackTrace()` is bad practice.

Could you update this PR to use @nharrand's solution?

**Comment by zielint0** [comment]
> We don't use one logger per class in Spoon.

Please explain motivation.

> Could you update this PR to use @nharrand's solution?

Take a look. Maybe this is what you meant.

File: src/main/java/spoon/decompiler/SpoonClassFileTransformer.java [hunk 1]
```diff
@@ -136,6 +133,7 @@ public SpoonClassFileTransformer(Predicate<String> classNameFilter,
 			ProtectionDomain protectionDomain,
 			byte[] classfileBuffer
 	) throws IllegalClassFormatException {
+		IncrementalLauncher launcher = new IncrementalLauncher(inputSources, classPath, cache);
```
Comment (nharrand) [inline src/main/java/spoon/decompiler/SpoonClassFileTransformer.java]: I think it is better to build the model only after filtering classes the user does not want to transform. (And after decompilation has been tried with succes, althought this last one is less important).

Comment (zielint0) [reply src/main/java/spoon/decompiler/SpoonClassFileTransformer.java]: If it is not a low hanging fruit, I leave it as if for now.

**Pull request closed without merging**